### PR TITLE
Change Battle Frontier validation to species flag

### PR DESCRIPTION
**Pull request by kittenchilly**
## Description
EDIT by Edu: This PR adds `isFrontierBanned` instead of splitting the legendary flag in two. 

~~I split the isLegendary flag into 2 flags that more closely respect how Pokemon are flagged in the games: isRestrictedLegendary and isSubLegendary.
These can be used for any custom stuff but without any of that currently they are used for:
isRestrictedLegendary must have 3 perfect IVs, and additionally can't be used in the Battle Frontier.
isSubLegendary must have 3 perfect IVs.~~

With this change, I also updated the Battle Frontier validity check to use the ~~isRestrictedLegendary and isMythical flags~~ `isFrontierBanned` flag for if a mon should be banned, rather than a separate hardcoded array. This should make updating the banlist easier, and also consequently fixes Terapagos and Pecharunt not being banned.

Also renamed isParadoxForm to isParadox since they aren't forms, and removed the isParadox flag from Koraidon and Miraidon since they are just legendaries.

## **Discord contact info**
kittenchilly

### CODE REVIEW

**Comment by kittenchilly** [comment]
Reviews addressed and modern fixed

File: src/frontier_util.c [hunk 1]
```diff
@@ -2021,17 +2001,26 @@ static void CheckPartyIneligibility(void)
...
         gSpecialVar_0x8004 = TRUE;
         count = 0;
-        for (i = 0; gFrontierBannedSpecies[i] != 0xFFFF; i++)
-            count = AppendCaughtBannedMonSpeciesName(gFrontierBannedSpecies[i], count, caughtBannedMons);
+        for (i = 0; i <= NATIONAL_DEX_COUNT; i++) // Loop to actually display the banned species' names (needs total count from the above)
+        {
+            species = NationalPokedexNumToSpecies(i);
+            if (gSpeciesInfo[species].isRestrictedLegendary || gSpeciesInfo[species].isMythical)
+                count = AppendCaughtBannedMonSpeciesName(species, count, caughtBannedMons);
+        }
```
Comment (AlexOn1ine) [inline src/frontier_util.c:2023]: This is a good idea but unfortunately it takes to long. Could you bring back the hardcoded list for now? Maybe we have a better idea in the future.

Comment (kittenchilly) [reply src/frontier_util.c:2023]: Alright sure, but only for this part since I think this is the one one where it matters.

Comment (kittenchilly) [reply src/frontier_util.c]: Alright, addressed

Comment (AsparagusEduardo) [reply src/frontier_util.c]: If the table is still being used in the dialog display, modifying the flags will not make the dialog update, which make it less user-friendly.

Comment (kittenchilly) [reply src/frontier_util.c]: > If the table is still being used in the dialog display, modifying the flags will not make the dialog update, which make it less user-friendly.

So what do we want to do? Just sacrifice the speed for userability?

**Comment by kittenchilly** [comment]
Reviews addressed again, should be fast now without using a hardcoded list

**Comment by AsparagusEduardo** [comment]
> Reviews addressed again, should be fast now without using a hardcoded list

"Should be"? I guess I gotta test it in-game myself if you aren't.

**Comment by kittenchilly** [comment]
> > "Should be"? I guess I gotta test it in-game myself if you aren't.
> 
> Sorry if this was too mean. After what happened with #4373, I got a bit on edge about not testing code that theorically works.

It's ok. I actually did test it and thought it was faster but I just wanted other people to judge for themselves, I guess. Sorry for the confusing language

File: src/party_menu.c [hunk 2]
```diff
@@ -6948,11 +6947,8 @@ static bool8 GetBattleEntryEligibility(struct Pokemon *mon)
         return TRUE;
     default: // Battle Frontier
         species = GetMonData(mon, MON_DATA_SPECIES);
-        for (; gFrontierBannedSpecies[i] != 0xFFFF; i++)
-        {
-            if (gFrontierBannedSpecies[i] == GET_BASE_SPECIES_ID(species))
-                return FALSE;
-        }
+        if (gSpeciesInfo[species].isFrontierBanned || gSpeciesInfo[species].isMythical)
```
Comment (AsparagusEduardo) [inline src/party_menu.c:6950]:
```suggestion
        if (gSpeciesInfo[species].isFrontierBanned)
```

**Review by AsparagusEduardo — CHANGES_REQUESTED** [review]
Forgot to remove the `isMythical` from the frontier functions ^^;

**Comment by kittenchilly** [comment]
> Forgot to remove the `isMythical` from the frontier functions ^^;

Done

**Comment by AsparagusEduardo** [comment]
> Done

Lol, I was about to do that too 😂

**Review by AsparagusEduardo — APPROVED** [review]

File: src/frontier_util.c [hunk 3]
```diff
@@ -2020,27 +1994,40 @@ static void CheckPartyIneligibility(void)
     if (numEligibleMons < toChoose)
     {
         s32 i;
-        s32 caughtBannedMons = 0;
-        s32 species = gFrontierBannedSpecies[0];
-        for (i = 0; species != 0xFFFF; i++, species = gFrontierBannedSpecies[i])
+        s32 baseSpecies = 0;
+        s32 totalCaughtBanned = 0;
+        s32 caughtBanned[100] = {0};
```
Comment (AlexOn1ine) [inline src/frontier_util.c:1999]: this should use u32's

File: src/party_menu.c [hunk 4]
```diff
@@ -6926,7 +6926,6 @@ static u8 GetPartySlotEntryStatus(s8 slot)
 
 static bool8 GetBattleEntryEligibility(struct Pokemon *mon)
 {
-    u16 i = 0;
     u16 species;
```
Comment (AlexOn1ine) [inline src/party_menu.c:6929]: u32

File: src/frontier_util.c (hunk 3 above)
Comment (AsparagusEduardo) [inline src/frontier_util.c:1999]:
```suggestion
        u32 baseSpecies = 0;
        u32 totalCaughtBanned = 0;
        u32 caughtBanned[100] = {0};
```

File: src/frontier_util.c [hunk 5]
```diff
@@ -2020,27 +1994,40 @@ static void CheckPartyIneligibility(void)
     if (numEligibleMons < toChoose)
     {
         s32 i;
```
Comment (AsparagusEduardo) [inline src/frontier_util.c:1996]:
```suggestion
        u32 i;
```

File: src/party_menu.c (hunk 4 above)
Comment (AsparagusEduardo) [inline src/party_menu.c:6929]:
```suggestion
    u32 species;
```

**Review by AsparagusEduardo — CHANGES_REQUESTED** [review]

**Review by AsparagusEduardo — APPROVED** [review]

**Pull request merged**